src/Asset.php

```diff
@@ -15,25 +13,29 @@
 
 interface Asset
 {
-
-    // location types
+    // Location types
     public const FRONTEND = 2;
     public const BACKEND = 4;
     public const CUSTOMIZER = 8;
     public const LOGIN = 16;
     public const BLOCK_EDITOR_ASSETS = 32;
-    // hooks
+    public const BLOCK_ASSETS = 64;
```

Comment:
i darkly renember why i did not add this. The `enqueue_block_assets` is a hook which is triggered in "editor interface" and "frontend". Actually the callback of this hook [triggers internally](https://github.com/WordPress/WordPress/blob/3c1961faa14994356dd772bf91d68384785d3e1e/wp-includes/script-loader.php#L2188) the `Asset::BLOCK_EDITOR_ASSETS` as well.

So this action is basically a shortcut/combination of `Asset::FRONTEND | Asset::BLOCK_EDITOR_ASSETS`. Should we then, instead of using "64" here just use 34 (FRONTEND + BLOCK_EDITOR_ASSETS)?


## More Context/Links
----

https://developer.wordpress.org/reference/hooks/enqueue_block_assets/
> Fires after enqueuing block assets for both **editor** and **front-end**.

https://github.com/WordPress/WordPress/blob/master/wp-includes/default-filters.php#L516-L517
([internally](https://github.com/WordPress/WordPress/blob/3c1961faa14994356dd772bf91d68384785d3e1e/wp-includes/script-loader.php#L2167) the callback is detecting if the `WP_Screen::is_block_editor`)

-----

https://developer.wordpress.org/reference/hooks/enqueue_block_editor_assets/
> Fires after block assets have been enqueued for the **editing interface**.

https://github.com/WordPress/gutenberg/blob/f8f86075e5c323a0e215545d595c48a6c3c08d37/lib/edit-site-page.php#L207

--> [`enqueue_block_editor_assets`](https://github.com/WordPress/gutenberg/blob/f8f86075e5c323a0e215545d595c48a6c3c08d37/lib/edit-site-page.php#L200)

Reply:
> So this action is basically a shortcut/combination of Asset::FRONTEND | Asset::BLOCK_EDITOR_ASSETS.

Yes, that's exactly what it is.

> So this action is basically a shortcut/combination of Asset::FRONTEND | Asset::BLOCK_EDITOR_ASSETS. Should we then, instead of using "64" here just use "34" (FRONTEND + BLOCK_EDITOR_ASSETS)?

Yes, we can do that. But I don't see much benefit. Also there's a bit of difference in the ordering, because  `enqueue_block_assets` happens before `wp_enqueue_assets` (in frontoffice) and `wp_admin_scripts` (in backoffice). Sometimes ordering is important for assets so if WordPress provides a separate hook I don't see why don't treat it separately. 

This is also future proof. Considering how fast Gutenberg changes, in a month or two `enqueue_block_assets`  could be used somewhere else and/or have a different meaning.

Reply:
Agree, let's keep it in as new Hook/Location. ✔️